BuildResidentialHPXML/resources/schedules.rb

```diff
@@ -524,12 +521,14 @@ def create_stochastic_schedules(args:)
     sink_flow_rate = gaussian_rand(prng, sink_flow_rate_mean, sink_flow_rate_std, 0.1)
     @total_days_in_year.times do |day|
       cluster_per_day.times do |cluster_count|
-        todays_probable_steps = sink_activtiy_probable_mins[day * @mkc_ts_per_day...((day + 1) * @mkc_ts_per_day)]
+        todays_probable_steps = sink_activity_probable_mins[day * @mkc_ts_per_day...((day + 1) * @mkc_ts_per_day)]
         todays_probablities = todays_probable_steps.map.with_index { |p, i| p * hourly_onset_prob[i / @mkc_ts_per_hour] }
         prob_sum = todays_probablities.sum(0)
         normalized_probabilities = todays_probablities.map { |p| p * 1 / prob_sum }
         cluster_start_index = weighted_random(prng, normalized_probabilities)
-        sink_activtiy_probable_mins[cluster_start_index] = 0 # mark the 15-min interval as unavailable for another sink event
+        if sink_activity_probable_mins[cluster_start_index] != 0
+          sink_activity_probable_mins[cluster_start_index] = 0 # mark the 15-min interval as unavailable for another sink event
+        end
```

Comment:
This reduces runtime by another 0.5s.

Reply:
Hmm, interesting. I wonder why setting a value at an index would be that expensive that putting a conditional guard would save time?  How are you measuring saved times?

Reply:
In general, the sink code did take significant time to run, but yes, I was also quite surprised to see that the above line of code was the culprit. But I reproduced it repeatedly. That said, feel free to test it yourself!

I basically used code like:
```
sum_t = 0
@total_days_in_year.times do |day|
      cluster_per_day.times do |cluster_count|
            ...
            t = Time.now
            sink_activity_probable_mins[cluster_start_index] = 0 # mark the 15-min interval as unavailable for another sink event
            sum_t += (Time.now - t)
      end
end
puts sum_t
```

Reply:
Thanks! It's kinda surprising that it works this way, but yeah, can't deny hard results :)